sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -84,15 +84,21 @@ class SessionCatalog(
   // ----------------------------------------------------------------------------
 
   def createDatabase(dbDefinition: CatalogDatabase, ignoreIfExists: Boolean): Unit = {
-    externalCatalog.createDatabase(dbDefinition, ignoreIfExists)
+    if (!databaseExists(dbDefinition.name)) {
+      externalCatalog.createDatabase(dbDefinition, ignoreIfExists)
+    }
   }
 
   def dropDatabase(db: String, ignoreIfNotExists: Boolean, cascade: Boolean): Unit = {
-    externalCatalog.dropDatabase(db, ignoreIfNotExists, cascade)
+    if (databaseExists(db)) {
+      externalCatalog.dropDatabase(db, ignoreIfNotExists, cascade)
+    }
```

Comment:
This is now wrong. We used to pass the flag `ignoreIfNotExists` to the external catalog, which will handle the exception throwing. Now we'll ignore this flag if the database doesn't exist.